bpf/mock/helpers.h

```diff
@@ -0,0 +1,84 @@
+/* SPDX-License-Identifier: GPL-2.0 */
```

Comment:
This is now generated, correct?

Reply:
Right!

Reply:
If it's generated, do we still need to check this one in?

Reply:
I would suggest we keep them because it could be possible that a future change on bpf/helpers.h may disable the script from generating the right headers (could be some lines wrongly trigger sed commends). If we keep them, at least developers can revise the headers manually.

Reply:
I think we should commit to either autogenerating or manually generating, not mixing the two. And my preference would be on autogenerating. Otherwise as soon as one manual change is introduced, the automatic generation piece is broken and useless.

I see the concern though, for instance how will someone in future understand that the headerfile is not being generated correctly? Can we make that more obvious somehow? I'm not sure, as I don't have a good sense for what it looks like for a developer to work on a new test and hit an issue where this autogenerated header doesn't work.

Reply:
Make sense, but I think there are another two reasons to keep them. First, the list of helper functions does not change often, so developers do not really need to generate the helper headers and generate the mock libraries for helpers, instead they can plug and play if we keep them. Second, even if the future add-ons cause some trouble, developers probably could use the old version mock libraries because they might not even need the newly added helpers.

Reply:
I agree with you.

> We can add tests for this into the CI so that any time the affected files are modified, we run specific mock generation + mock compile tests to validate that things aren't randomly breaking for developers later on when they try to write tests.

The problem is that it is hard to tell if those 'add-ons' will break the script to generate headers. We can assume that if cmock can successfully generate mock library for the header, then the 'add-ons' are safe?